spec-dtslint/operators/tap-spec.ts

```diff
@@ -0,0 +1,20 @@
+import { of } from 'rxjs';
+import { tap } from 'rxjs/operators';
+
+it('should infer correctly', () => {
+  const a = of(1, 2, 3).pipe(tap()); // $ExpectType Observable<number>
+});
+
+it('should accept partial observer', () => {
+  const a = of(1, 2, 3).pipe(tap({ next: (x: number) => x })); // $ExpectType Observable<number>
+  const b = of(1, 2, 3).pipe(tap({ error: (x: number) => x })); // $ExpectType Observable<number>
```

Comment:
The `error` callbacks should be typed as `any`. There is no guarantee that the callback will be passed a particular type. Also, the error callback should not return a value. Use `error: (err: any) => {}`.